src/context/GameContext.js

```diff
@@ -0,0 +1,96 @@
+import { useState } from 'react';
+import { createContext } from 'react';
+
+const GameContext = createContext();
+
+const winStates = [
+  [0, 1, 2],
+  [3, 4, 5],
+  [6, 7, 8],
+  [0, 3, 6],
+  [1, 4, 7],
+  [2, 5, 8],
+  [0, 4, 8],
+  [2, 4, 6],
+];
+
+const GameProvider = ({ children }) => {
+  const [gameBoard, setGameBoard] = useState(['', '', '', '', '', '', '', '', '']);
+  const [currentPlayer, setCurrentPlayer] = useState('X');
+  const [gameMessage, setGameMessage] = useState('Your Turn X');
+  const [gameIsActive, setGameIsActive] = useState(true);
+
+  function handleResetClick() {
+    setGameBoard(['', '', '', '', '', '', '', '', '']);
+    setCurrentPlayer('X');
+    setGameMessage('Your Turn X');
+    setGameIsActive(true);
+  }
+
+  function handleTileClick(id) {
+    // Guard clause
+    if (!gameIsActive) return;
+    if (gameBoard[id] !== '') return;
+
+    // Update gameBoard with new tile value
+    const newGameBoard = [...gameBoard];
+    newGameBoard[id] = currentPlayer;
+    setGameBoard(newGameBoard);
```

Comment:
i THINK (might be wrong) you could write this as ```setGameBoard((prevState) => [...prevState, id: currentPlayer])``` or maybe ```prevState[id]: currentPlayer``` not that it would be better or worse just riffin